dfusion_rust_core/src/models/util.rs

```diff
@@ -32,4 +33,78 @@ pub fn to_value<T: ToString>(value: &T) -> Value {
     Value::from(
         BigDecimal::from_str(&value.to_string()).unwrap()
     )
+}
+
+pub trait EntityParsing {
+    fn from_entity(entity: &Entity, field: &str) -> Self;
+}
+
+impl EntityParsing for u8 {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        u8::try_from(entity
+            .get(field)
+            .and_then(|value| value.clone().as_int())
+            .unwrap_or_else(|| panic!("Couldn't get field {} as uint", field))
+        ).unwrap_or_else(|_| panic!("Couldn't cast {} from i32 to u8", field))
+    }
+}
+
+impl EntityParsing for u16 {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        u16::try_from(entity
+            .get(field)
+            .and_then(|value| value.clone().as_int())
+            .unwrap_or_else(|| panic!("Couldn't get field {} as uint", field))
+        ).unwrap_or_else(|_| panic!("Couldn't cast {} from i32 to u16", field))
+    }
+}
+
+impl EntityParsing for u128 {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        u128::from_str(&entity
+            .get(field)
+            .and_then(|value| value.clone().as_big_decimal())
+            .map(|decimal| decimal.to_string())
+            .unwrap_or_else(|| panic!("Couldn't get field {} as big decimal", field))
+        ).unwrap_or_else(|_| panic!("Couldn't cast {} from string to u128", field))
+    }
+}
+
+impl EntityParsing for U256 {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        U256::from_str(&entity
+            .get(field)
+            .and_then(|value| value.clone().as_big_decimal())
+            .map(|decimal| decimal.to_string())
+            .unwrap_or_else(|| panic!("Couldn't get field {} as big decimal", field))
+        ).unwrap_or_else(|_| panic!("Couldn't cast {} from string to U256", field))
+    }
+}
+
+impl EntityParsing for H256 {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        H256::from_str(&entity
+            .get(field)
+            .and_then(|value| value.clone().as_string())
+            .unwrap_or_else(|| panic!("Couldn't get field {} as string", field))
+        ).unwrap_or_else(|_| panic!("Couldn't cast {} from string to H256", field))
+    }
+}
+
+impl EntityParsing for Vec<u128> {
+    fn from_entity(entity: &Entity, field: &str) -> Self {
+        entity
+            .get(field)
+            .and_then(|value| value.clone().as_list())
+            .map(|list| list
+                .into_iter()
+                .map(|value| u128::from_str(
```

Comment:
Can we reuse the from_entity code for u128 here?

Reply:
It would have to be from_value, since the item is a `Vec<Value>`.